client/pooling.zig

```diff
@@ -0,0 +1,179 @@
+const std = @import("std");
+
+const log = std.log.scoped(.pooling);
+
+pub fn SimpleKey(comptime Child: type) type {
+    return struct {
+        value: Child,
+
+        const Self = @This();
+
+        pub fn eql(self: Self, other: Self) bool {
+            return self.value == other.value;
+        }
+    };
+}
+
+pub fn SizedKey(comptime Child: type) type {
+    return struct {
+        value: Child,
+        size: usize,
+
+        const Self = @This();
+
+        pub fn eql(self: Self, other: Self) bool {
+            return self.value == other.value;
+        }
+
+        pub fn fits(self: Self, other: Self, smallest_size: usize) bool {
+            return self.size >= other.size and self.size < smallest_size;
+        }
+
+        pub fn exactMatch(self: Self, other: Self) bool {
+            return self.size == other.size;
+        }
```

Comment:
I would prefer the pool API to have a single compare function that acts like this:

```suggestion
        pub fn compare(self: Self, other: Self) std.math.Order {
            if(left.value != right.value or left.size > right.size) {
            	return .lt; // less than means it won't fit
            }
            
            if(left.size == right.size) {
            	return .eq; // equals means it's an exact match
            }
            
            if(left.size < right.size) {
            	return .gt; // greater than means it's a match, but not a perfect match
            }
        }
```

For the simpler case, it can be implemented as such:
```zig
        pub fn compare(self: Self, other: Self) std.math.Order {
            return if(self == other) .eq else .lt;
        }
```